Guide.docc/IncrementalAdoption.md

```diff
@@ -227,3 +227,98 @@ NS_SWIFT_ASYNC_NAME
 NS_SWIFT_ASYNC_NOTHROW
 NS_SWIFT_UNAVAILABLE_FROM_ASYNC(msg)
 ```
+
+### Dealing mistakes in isolation annotations in Objective-C libraries
+
+While the SDKs and other objective-c libraries make progress in adopting Swift concurrency,
+they will often go through the exercise of codifying contracts which were only explained in
+documentation, like "this will always be called on the main thread", and turn them intro 
+compile and runtime enforced isolation checks that Swift will then perform when you adopt such APIs.
+
+For example, the fictional `NSJetPack` protocol generally invokes all of its delegate methods
+on the main thread, and is MainActor-isolated. The SDK author therefore annotates the type like this:
+
+```swift
+NS_SWIFT_UI_ACTOR // SDK author annotated using MainActor in recent SDK audit
+@protocol NSJetPack // fictional protocol
+  // ...
+@end
+```
+
+As such, all member methods of this protocol inherit the `@MainActor` annotation, and for most methods this is correct.
+However, in this example, let us consider a method which was previously documented as follows:
+
+```swift
+NS_SWIFT_UI_ACTOR // SDK author annotated using MainActor in recent SDK audit
+@protocol NSJetPack // fictional protocol
+/* Return YES if this jetpack supports flying at really high altitude!
+ 
+ JetPackKit invokes this method at a variety of times, and not always on the main thread. For example, ...
+*/
+@property(readonly) BOOL supportsHighAltitude;
+
+@end
+```
+
+This method accidentally inherited the `@MainActor` property from the enclosing type,
```

Comment:
I've been trying to be really careful about my use of "inherit" vs "infer" when it comes to isolation. Is "inherit" the right term here?